typescript-tests/Entity.test.ts

```diff
@@ -0,0 +1,337 @@
+import { denormalize, IdFunction, normalize, schema } from '../src';
```

Comment:
Why are all of these test files complete duplicates of the source test files? I would prefer to not need to maintain duplicates like this.

Reply:
Since the test files verify that all of the functionality works as expected, duplicating the files and having the typescript compiler run on them was the simplest way to verify that all of the functionality works as expected within typescript as well. There was actually one test that did have to change, in Array.test.ts `denormalizes multiple entities`: The entry of `{ id: { id: '789' ), schema: 'dogs' }` wasn't valid given the types that I assigned, and it didn't look like it actually should be, based on the usage of denormalize.

Another option would be to remove the javascript tests, leaving only the typescript ones for most functionality. That would test both that the functionality works and that the given type definitions are correct. The only things that would need to be added at that point are tests that expect to throw an error in javascript, but wouldn't compile in typescript (e.g. passing `42` to the entity key).

I'm happy to make that change if you'd like; I just didn't want to remove existing tests on my own.

Reply:
@paularmstrong I've removed the duplicate tests. This also has the advantage that you can now clearly see what's possible in javascript but not in typescript with the new definitions file.